ingestionpolicies.go

```diff
@@ -103,7 +103,45 @@ func (p *IngestionPolicies) Create(policy *IngestionPolicyRequest) (*IngestionPo
 	return &ingestionPolicy, nil
 }
 
-func (p *IngestionPolicies) Update(policy *IngestionPolicyRequest) (*IngestionPolicyResponse, error) {
+func (p *IngestionPolicies) Update(policy *IngestionPolicyResponse) error {
+
+	var accounts []string
+	for _, v := range policy.Accounts {
+		accounts = append(accounts, v.ID)
+	}
+
+	var groups []string
+	for _, v := range policy.Groups {
+		groups = append(groups, v.ID)
+	}
+
+	var pointTags []map[string]string
+	for _, v := range policy.Tags {
+		pointTags = append(pointTags, map[string]string{"Key": v.Key, "Value": v.Value})
+	}
+
+	policyRequest := &IngestionPolicyRequest{
+		ID:          policy.ID,
+		Name:        policy.Name,
+		Description: policy.Description,
+		Scope:       policy.Scope,
+		Accounts:    accounts,
+		Groups:      groups,
+		Sources:     policy.Sources,
+		Namespaces:  policy.Namespaces,
+		Tags:        pointTags,
+	}
+
+	policy, err := p.RawUpdate(policyRequest)
```

Comment:
lint flagged this line. In go, parameters are always pass-by-value, so the caller won't see this change to the policy pointer.  There are a few things you can do:

```
newPolicy, err := p.RawUpdate(policyRequest)
if err != nil {
    return err
}
*policy = *newPolicy
return nil
```

Doing this will modify the passed in policy in place.

Reply:
My IDE flagged this too. I haven't done C++ in quite a while so definitely need some help with ensuring I am managing them properly. Thanks for the example!